src/content/tools/hooks.md

```diff
@@ -0,0 +1,377 @@
+        📄 example_test.dart
+```
+
+## Hooks
+
+Currently, the build hook and link hook are available,
+but more are planned. To learn more about these hooks,
+see the following sections.
+
+### Build hooks
+
+A build hook builds assets that you want
+to run during your project's build process. This is useful
+for tasks that need to happen before your main
+application code is compiled, such as:
+
+* Compiling C, C++, Rust, or other native code into a
+  dynamic library.
+* Downloading or generating other assets required at
+  runtime.
+
+Use the [`build`][] function to create a build hook.
+Build hooks are stored in the `hook/build.dart` directory
+and create code assets in the `build/` directory.
+
+[`build`]: {{site.pub-api}}/hooks/0.19.5/hooks/build.html
+
+### Link hooks
+
+A link hook shrinks assets to only the
+required parts based on Dart AOT tree-shaking information.
+
+Use the [`link`][] function to create a build hook.
+Link hooks are stored in the `hook/link.dart` directory
+and create code assets and data assets in the
+`build/` directory.
+
+[`link`]: {{site.pub-api}}/hooks/0.19.5/hooks/link.html
+
+## Assets
+
+Assets are the files output from a hook that need to be
+bundled in a Dart application. Assets can be accessed
+at runtime from the Dart code. Currently, the Dart SDK can
+use the `CodeAsset` and `DataAsset` asset types.
+
+### CodeAsset type
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+A code asset can be used with build and link hooks.
+
+[`CodeAsset`]: {{site.pub-api}}/code_assets/latest/code_assets/CodeAsset-class.html
+
+### DataAsset type
+
+A [`DataAsset`][] represents a data asset in bytes format.
+It can be accessed by Dart at runtime and used in
+Dart standalone packages. `DataAsset` is part of the
+`data_asset` package and can be used with link hooks.
+
+[`DataAsset`]: {{site.pub-api}}/data_assets/latest/data_assets/DataAsset-class.html
+
+## Use a hook
+
+To setup a hook, use a hook to generate assets, and use
+those assets in your Dart project, see the steps in the
+following sections.
+
+### Add dependencies
+
+Open your package in a code editor, navigate to its
+`pubspec.yaml`, and then add the following dependencies as
+needed under `dependencies`:
+
+* `hooks`
+* `code_assets` (for build hooks and link hooks)
+* `data_assets` (for link hooks)
+
+You will also likely need to add a tool for your specific
+task, such as `native_toolchain_c` for compiling C code.
+
+:::note
+You need to add the dependencies under `dependencies`, not
+`dev_dependencies`. The hooks will be run by packages and
+applications depending on your package, so the
+Dart code needs to be part of the resolution in those
+packages.
+:::
+
+Example dependencies for a build hook:
```

Comment:
It doesn't make sense to use link hooks without a build hook. The link hook is
meant to make assets from a build hook smaller.

Also, both build and link hooks can work with code assets. (And once data assets
support is added both hooks can work with data assets.)